python/paddle/v2/framework/evaluator.py

```diff
@@ -1,59 +1,134 @@
-import paddle.v2.framework.op as op
-import numpy as np
+from paddle.v2.framework.framework import Program, g_program, unique_name
+from paddle.v2.framework.layer_helper import LayerHelper
 import paddle.v2.framework.core as core
 
 
-def avg_accumulate(accumulated_var, per_eval, num_batches, place):
-    t = np.array(accumulated_var.get_tensor())
-    t[0] += per_eval[0]
-    accumulated_var.get_tensor().set([t[0] / float(num_batches)], place)
+class Evaluator(object):
+    """
+    Evalutor Base class.
 
+    create metric states
+    add mini-batch evaluator caculate operator
+    add increment operator to accumulate the metric states
+    """
 
-class Evaluator(object):
-    def __init__(self,
-                 scope,
-                 operator='accuracy',
-                 input='Inference',
-                 label='Label',
-                 output='Output',
-                 place=core.CPUPlace()):
+    def __init__(self, name, **kwargs):
+        self._states = {}
+        self._helper = LayerHelper(layer_type=name, **kwargs)
+        # if kwargs.has_key("program"):
+        #     self._program =  kwargs.get("program")
+        # else:
+        #     self._program = g_program
+
+    # def _update(self):
+    #     """
+    #     Updates the internal states througth operator
+    #   """
+    #     raise NotImplementedError()
+
+    def reset(self, executor, program=None):
```

Comment:
Do not expose reset_program to the upper level. It should be created, run and destroyed in the `reset` function.

Reply:
Done.